cpp/modules/luma.gl/core/src/model.h

```diff
@@ -68,30 +74,43 @@ class Model {
   wgpu::ShaderModule fsModule;
 
  private:
-  std::shared_ptr<wgpu::Device> _device;
+  wgpu::Device _device;
```

Comment:
Nits:
- I think it is nice that anyone can get the device from a model etc by just calling `model->device`
- I don't think everything needs to be private. 
- I can live with the minor risk that someone assigns to it.

For the others it probably makes sense to have them private.